ccmlib/utils/sni_proxy.py

```diff
@@ -0,0 +1,152 @@
+                                             clientKeyData=key_data,
+                                             username=username,
+                                             password=password,
+                                             insecureSkipTlsVerify=True)},
+                  contexts={'default': dict(datacenterName='eu-west-1', authInfoName='default')},
+                  currentContext='default')
+
+    with open(os.path.join(ssl_dir, 'config_data.yaml'), 'w') as config_file:
+        config_file.write(yaml.safe_dump(config, sort_keys=False))
+
+    config = dict(datacenters={'eu-west-1': dict(certificateAuthorityPath=os.path.join(ssl_dir, 'ccm_node.cer'),
+                                                 server=f'{host}:{port}',
+                                                 nodeDomain='cluster-id.scylla.com')},
+                  authInfos={'default': dict(clientCertificatePath=os.path.join(ssl_dir, 'ccm_node.cer'),
+                                             clientKeyPath=os.path.join(ssl_dir, 'ccm_node.key'),
+                                             username=username,
+                                             password=password,
+                                             insecureSkipTlsVerify=True)},
+                  contexts={'default': dict(datacenterName='eu-west-1', authInfoName='default')},
+                  currentContext='default')
+
+    with open(os.path.join(ssl_dir, 'config_path.yaml'), 'w') as config_file:
+        config_file.write(yaml.safe_dump(config, sort_keys=False))
+
+    return os.path.join(ssl_dir, 'config_data.yaml'), os.path.join(ssl_dir, 'config_path.yaml')
+
+
+def stop_sni_proxy(docker_id):
+    subprocess.check_output(['/bin/bash', '-c', f'docker rm -f {docker_id}'])
+
+
+def start_sni_proxy(conf_dir, nodes_info, listen_port=443):
+    from textwrap import dedent
+    sniproxy_conf_tmpl = dedent("""
+    user sniproxy
+    pidfile /var/run/sniproxy/sniproxy.pid
+    
+    error_log {
+      filename /dev/stderr
+      priority debug
+    }
+    
+    listener $FIRST_ADDRESS $listen_port {
+      proto tls
+      access_log {
+        filename /dev/stdout
+      }
+    }
+    
+    table {
+    $TABLES
+    }
+    """)
+    tables = ""
+    mapping = {}
+    address, port, host_id = list(nodes_info)[0]
+    tables += f"  any.cluster-id.scylla.com {address}:{port}\n"
+    mapping['FIRST_ADDRESS'] = address
+    mapping['listen_port'] = listen_port
+
+    for address, port, host_id in nodes_info:
+
+        tables += f"  {host_id}.cluster-id.scylla.com {address}:{port}\n"
+
+    tmpl = string.Template(sniproxy_conf_tmpl)
+    sniproxy_conf_path = os.path.join(conf_dir, 'sniproxy.conf')
+
+    with open(sniproxy_conf_path, 'w') as fp:
+        fp.write(tmpl.substitute(TABLES=tables, **mapping))
+
+    docker_id = subprocess.check_output(['/bin/bash', '-c', f'docker run -d --network=host -v {sniproxy_conf_path}:/etc/sniproxy.conf -p 443 -it sniproxy'], universal_newlines=True)
```

Comment:
Where is this `sniproxy` image from?

Reply:
oops, I've forgot to put in the build part of the image

```
FROM alpine

RUN apk --no-cache add sniproxy

EXPOSE 80 443
ENTRYPOINT ["sniproxy", "-f"]
```

Reply:
I've pushed a the dockerfile, and a few fixes